modules/internal/ChapelArray.chpl

```diff
@@ -2939,13 +2991,189 @@ module ChapelArray {
+    if ranges.size == 1 && ranges(0).bounds == boundKind.low {
+      return arr.chpl_aliasReshape({ranges(0).low..#arr.size}, false);
+    } else {
+      return arr.chpl_aliasReshape({(...ranges)}, checkReshapeDimsByDefault);
+    }
+  }
+
+  pragma "no promotion when by ref"
+  pragma "fn returns aliasing array"
+  @chpldoc.nodoc
+  @edition(first="pre-edition")
+  proc reshape(arr: [], ranges: range(?)..., checkDims: bool) {
+    if ranges.size == 1 && ranges(0).bounds == boundKind.low {
+      return arr.chpl_aliasReshape({ranges(0).low..#arr.size}, false);
+    } else {
+      return arr.chpl_aliasReshape({(...ranges)}, checkDims);
+    }
+  }
+
+  pragma "no promotion when by ref"
+  pragma "fn returns aliasing array"
+  pragma "last resort"
+  @chpldoc.nodoc
+  @edition(first="pre-edition")
+  proc reshape(arr: [], ranges: range(?)..., param copy: bool) {
+    if copy {
+      return arr.chpl_copyReshape({(...ranges)}, checkReshapeDimsByDefault);
+    } else {
+      if ranges.size == 1 && ranges(0).bounds == boundKind.low {
+        return arr.chpl_aliasReshape({ranges(0).low..#arr.size}, false);
+      } else {
+        return arr.chpl_aliasReshape({(...ranges)}, checkReshapeDimsByDefault);
+      }
+    }
+  }
+
+  pragma "no promotion when by ref"
+  pragma "fn returns aliasing array"
+  @chpldoc.nodoc
+  @edition(first="pre-edition")
+  proc reshape(arr: [], ranges: range(?)...,
+               checkDims = checkReshapeDimsByDefault, param copy = false) {
+    if copy {
+      return arr.chpl_copyReshape({(...ranges)}, checkDims);
+    } else {
+      if ranges.size == 1 && ranges(0).bounds == boundKind.low {
+        return arr.chpl_aliasReshape({ranges(0).low..#arr.size}, false);
+      } else {
+        return arr.chpl_aliasReshape({(...ranges)}, checkDims);
+      }
+    }
+  }
+
+  // This version takes a domain
+
+  pragma "no promotion when by ref"
+  pragma "fn returns aliasing array"
+  @chpldoc.nodoc
+  @edition(first="pre-edition")
+  proc reshape(arr: [], dom: domain(?), checkDims=checkReshapeDimsByDefault, param copy = false) {
+    if copy {
+      if !dom.isRectangular() then
+        compilerError("reshape() with copying is currently only supported for rectangular domains");
+      if boundsChecking then
+        chpl__validateReshape(arr, dom);
+      // TODO: Add a parallel linearize() iterator to all rectangular types
+      // and zip those instead of using this serial implementation
+      var B: [dom] arr.eltType = for (i,a) in zip(dom, arr) do a;
+      return B;
+    } else {
+      return arr.chpl_aliasReshape(dom, checkDims);
+    }
+
+  }
+
+  //
+  // I tried to just make the reshape() overloads above just contain
+  // the logic in the following two methods rather than using these
+  // helpers, but seemingly, something about returning an array view
+  // from a procedure that calls another procedure to create it didn't
+  // work (?).  Though I also wasn't easily able to reproduce this in
```

Comment:
Ah, sorry. Ignore my last, then. That's frustrating. Just old compiler things...